### PR TITLE
Add an ability to link dynamicaly with CUDA toolkit

### PR DESCRIPTION
**Pull request by JanuszL**
Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

## Description
- [ ] **Bug fix** (*non-breaking change which fixes an issue*)
- [x] **New feature** (*non-breaking change which adds functionality*)
- [ ] **Breaking change** (*fix or feature that would cause existing functionality to not work as expected*)
- [ ] **Refactoring** (*Redesign of existing code that doesn't affect functionality*)
- [ ] **Other** (*e.g. Documentation, Tests, Configuration*)

#### Additional information
- Affected modules and functionalities:

- Key points relevant for the review:

### Tests
- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -39,6 +39,7 @@ check_cxx_compiler_flag(-fopenmp-simd CXX_HAVE_OMP_SIMD)
 # Build options
 option(BUILD_DALI_NODEPS "Disable components that require extra external libraries to be present in the system. Effectively, it builds only the DALI core and kernel libraries")
 option(LINK_DRIVER "Links directly with libcuda.so instead of dlopen it at runtime" OFF)
+option(LINK_CUDA_DYNAMICALLY "Links CUDA libraires dynamically (NPP, nvJPEG, cuFFT)" OFF)
```
Comment (prak-nv) [inline CMakeLists.txt:42]: WITH_DYNAMIC_CUDA_LIBS would be in convention how it is usually called. Just a nitpick

File: dali/python/nvidia/dali/backend.py [hunk 2]
```diff
@@ -47,3 +47,7 @@ def deprecation_warning(what):
 
     for lib in default_plugins:
         LoadLibrary(os.path.join(os.path.dirname(__file__), lib))
+
+if CheckCudaVersion() == -1:
+    deprecation_warning("No CUDA installation detected. Please make sure it is available "
```
Comment (prak-nv) [inline dali/python/nvidia/dali/backend.py:52]: Should it really be just warning?

Comment (JanuszL) [reply dali/python/nvidia/dali/backend.py:52]: Yes. DALI can run without GPU if you use only CPU ops and device_id is None.

File: CMakeLists.txt (hunk 1 above)
Comment (JanuszL) [reply CMakeLists.txt:42]: Will fix.

**Comment by JanuszL** [comment]
Moved to https://github.com/NVIDIA/DALI/pull/3519

**Pull request closed without merging**